py/kubeflow/testing/cleanup_ci.py

```diff
@@ -608,9 +611,30 @@ def cleanup_certificates(args):
       # don't tell us what the hostname is. gcloud returns the host though
       # so the information should be somewhere. Maybe we just need a newer
       # version of the API?
+      #
+      # If we decode the pem it should be inter
       name = d["name"]
-      if age > datetime.timedelta(days=7):
-        logging.info("Deleting certifcate: %s", d["name"])
+
+      if not "certificate" in d:
+        logging.warning("Certificate %s is missing certificate", name)
+        continue
+
+      raw_certificate = d["certificate"]
+
+      cert = x509.load_pem_x509_certificate(raw_certificate.encode('utf-8'), default_backend())
+
+      # TODO(jlewi): Is there a way to do this without accessing protected attributes
+      domain = cert.subject._attributes[0]._attributes[0].value # pylint: disable=protected-access
```

Comment:
Are two _attributes[0] intended?
What's the format of subject?

Reply:
Yeah its a bit of a hack I couldn't figure out what the proper way to do this was which is why I'm accessing private attributes.
It ends up being the domain e.g jlewi-0-6-2-rc.endpoints.jlewi-dev.cloud.goog